Feature/home UI

joowon-jang opened the pull request:
## PR 유형

- [x] Feat: 새로운 기능 추가
- [ ] Fix: 버그 수정
- [ ] Design: CSS 등 사용자 UI 디자인 변경
- [ ] Style: 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] Refactor: 코드 리팩토링
- [ ] Comment: 주석 추가 및 수정
- [ ] Docs: 문서 수정
- [ ] Test: 테스트 추가, 테스트 리팩토링
- [ ] Chore: 빌드 부분 혹은 패키지 매니저 수정
- [ ] Rename: 파일 혹은 폴더명 수정
- [ ] Remove: 파일 혹은 폴더 삭제

## 🔧 반영 브랜치

<!-- ex) feature/login -> develop -->
feature/HomeUI -> develop

## 💬 Description

<!-- 리뷰하는 사람이 봐야할 내용을 작성해주세요. -->
<!-- 구현 한 기능에 대해 작성해 주세요. -->

- SearchForm: 검색창 컴포넌트
- Screen Reader 사용자에게 정보를 제공하기 위한 SrOnly css 추가했습니다.

## 📷 Screenshots

<!--스크린샷으로 보여줄 수 있는 이미지가 있다면 첨부해주세요!-->
<!-- 스크린샷은 선택사항입니다. -->
<img width="949" alt="스크린샷 2025-02-03 오후 4 35 06" src="https://github.com/user-attachments/assets/22bd76a9-e947-41e5-9240-cf87a8047e46" />
<img width="386" alt="스크린샷 2025-02-04 오후 3 58 53" src="https://github.com/user-attachments/assets/605b97a0-6b18-47f5-b908-fdf443475858" />

<!--
맨 마지막 줄에 아래 이슈
Fixes: 이슈 수정중 (아직 해결되지 않은 경우)
Resolves: 이슈를 해결했을 때 사용
Ref: 참고할 이슈가 있을 때 사용
Related to: 해당 커밋에 관련된 이슈번호 (아직 해결되지 않은 경우)

-여러 이슈를 입력시 comma(,) 단위로 구분해주세요
ex) Close #10, Resolves #123 -->
Resolves #16

joowon-jang committed:
build: framer-motion 설치

joowon-jang committed:
feat: CategoryButton의 WAI-ARIA 속성 추가

- aria-selected

joowon-jang committed:
chore: 홈 화면의 목록이 없을 때 보여줄 이미지 추가 (no_gift_image.webp)

joowon-jang committed:
rename: ConList -> ConListItem

- 추가로 border-top, hover 속성도 삭제

joowon-jang committed:
feat: globalStyles에 SrOnly css추가

joowon-jang committed:
feat: Home 페이지의 SearchForm 컴포넌트 제작

joowon-jang committed:
design: Header 컴포넌트 스타일 수정

joowon-jang committed:
design: globalStyles에서 body의 오버스크롤 방지

joowon-jang committed:
feat: HomePage의 전체적인 스타일링

- +(기프티콘 등록) 버튼 추가 예정

joowon-jang committed:
Merge branch 'develop' of https://github.com/FRONT-END-BOOTCAMP-PLUS-3/YogiCon into feature/HomeUI

joowon-jang committed:
modify: SearchForm의 border rem -> px 변경

joowon-jang committed:
design: CategoryButton의 color를 black으로 변경

- 모바일 접속하면 파란색으로 보여서 변경함

joowon-jang committed:
feat: home 페이지 UI 완성

yezzero commented on app/(home)/page.tsx:
```diff
@@ -1,25 +1,101 @@
+  top: 1.5rem;
+  z-index: 9998;
+  background-color: var(--white);
+  border-bottom: 0.1px solid #e7e7e7;
```
회색이 종류별로 globalstyles에 정의되어 있어요! 다른 것들에도 고거 쓰도록 수정하면 좋을 것 같습니덩

yezzero commented on app/(home)/page.tsx:
```diff
@@ -1,25 +1,101 @@
+                <ConListItem key={index} {...item} />
+              )
+          )}
+        </ConList>
```
여기에 아래 네비바와 1px정도 공간 차이가 있어서 이렇게 못생기게 되는데, 리스트가 끝나서 더이상 없음을 알려주는 문구를 넣는 거 어떨까요?? 아님 말구용
![image](https://github.com/user-attachments/assets/437c2493-5082-4348-bee4-7dbd3bf0ddad)

yezzero commented on app/(home)/components/SearchForm.tsx:
```diff
@@ -0,0 +1,77 @@
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: absolute;
```
버튼의 position을 absolute로 해서 이렇게 검색어가 길어지면 버튼을 덮어버리네,,요,, SearchInputWrapper에 justify-content: space-around를 넣는 것은 어떠할까요..
![image](https://github.com/user-attachments/assets/9962cc17-36b4-4b89-a935-a72799c11cf2)

joowon-jang replied:
오.. 역시 CSS고..수...
이따 고칠게요!

joowon-jang committed:
style: 컨벤션 변경에 따라 HomeLayout -> HomeContainer로 이름 변경

joowon-jang committed:
design: globalStyles 수정

joowon-jang committed:
design: LogoHeader의 padding 수정

joowon-jang committed:
design: SearchForm 디자인 수정

joowon-jang committed:
refactor: home에서 기프티콘 목록이 없을 때 화면 추가

- 서버에서 목록을 불러올 것을 고려해 conList를 state 및 useEffect로 관리
- 목록이 없을 때의 상황을 추가함에 따라 css 변경

joowon-jang committed:
modify: ConListItem의 props를 name -> product_name으로 변경

- 편의상 DB의 변수명과 일치하도록 변경함

joowon-jang committed:
design: home에서 기프티콘 목록이 없을 때의 이미지 스타일 추가 및 기타 수정사항

- NoConListText -> NoConText 이름 변경
- 등록하기 버튼 그림자 추가

joowon-jang committed:
feat: home에서 ConListItem 클릭 시 해당 Item의 view-con 페이지로 이동

joowon-jang committed:
fix: RootLayout과 Header에서 경로를 처리하던 부분 수정

- params를 받아서 사용하는 경로의 경우 처리가 안되고 있었음

joowon-jang committed:
feat: home의 conList를 api 요청 형태로 변경

- route.ts의 GET 사용

joowon-jang committed:
modify: home에서 useEffect 중복 사용 제거

joowon-jang committed:
bugFix: 가로폭 480px 이하 뷰포트에서 전체 font-size가 생각보다 작아지는 오류 수정

- globalStyles에서 html에만 media 규칙 적용하도록 변경

joowon-jang committed:
design: Header의 max-width: 768px

joowon-jang committed:
style: 서버와의 통신에 필요한 변수 이름 변경

- snake_case -> camelCase

joowon-jang committed:
feat: HomeUI 검색 기능

joowon-jang committed:
rename: Categories type 파일명 PascaCase로 변경

joowon-jang committed:
design: CategoryButton의 줄바꿈 현상 수정

joowon-jang committed:
Delete types/categories.ts

joowon-jang committed:
design: BackButton 스타일 변경

yezzero commented on app/(home)/page.tsx:
```diff
@@ -1,25 +1,185 @@
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
```
justify-content: flex-start 해야할 것 같습니다 리스트가 몇 개 없을 때 요소들이 중앙에 그냥 정렬되어버려요!

yezzero commented on app/(home)/page.tsx:
```diff
@@ -1,25 +1,185 @@
+  text-align: center;
+  font-size: 1.5rem;
+  font-weight: bold;
+  line-height: 44px; /* 183.333% */
```
px이 아니라 1.833 이렇게만 해주시면 될거같아요!

joowon-jang committed:
fix: 피드백 반영

- ConList 목록이 적을 때, 가운데에 나오는 현상 수정
- 검색결과가 없을 때, 빈화면이 나오는 현상 수정
- px -> rem 수정

yezzero reviewed (approved):
rrr
